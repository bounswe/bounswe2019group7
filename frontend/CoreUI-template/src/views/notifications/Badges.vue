<template>
  <div class="wrapper">
    <div class="animated fadeIn">
      <b-row>
        <b-col cols="12" md="6">
          <b-card
            header-tag="header"
            footer-tag="footer">
            <div slot="header">
              <i class="fa fa-align-justify"></i><strong> Bootstrap Badge</strong>
              <div class="card-header-actions">
                <a href="https://bootstrap-vue.js.org/docs/components/badge" class="card-header-action" rel="noreferrer noopener" target="_blank">
                  <small class="text-muted">docs</small>
                </a>
              </div>
            </div>
            <div>
              <h2>Example heading <b-badge>New</b-badge></h2>
              <h3>Example heading <b-badge>New</b-badge></h3>
              <h4>Example heading <b-badge>New</b-badge></h4>
              <h5>Example heading <b-badge>New</b-badge></h5>
              <h6>Example heading <b-badge>New</b-badge></h6>
            </div>
            <div slot="footer">
              <b-button variant="primary">
                Notifications <b-badge variant="light" :style="{position: 'static'}">4</b-badge>
              </b-button>
            </div>
          </b-card>
        </b-col>
        <b-col cols="12" md="6">
          <b-card
            header-tag="header"
            footer-tag="footer">
            <div slot="header"><i class='fa fa-align-justify'></i> Badge <small>contextual variations</small></div>
            <div>
              <b-badge class="m-1" variant="primary">Primary</b-badge>
              <b-badge class="m-1" variant="secondary">Secondary</b-badge>
              <b-badge class="m-1" variant="success">Success</b-badge>
              <b-badge class="m-1" variant="danger">Danger</b-badge>
              <b-badge class="m-1" variant="warning">Warning</b-badge>
              <b-badge class="m-1" variant="info">Info</b-badge>
              <b-badge class="m-1" variant="light">Light</b-badge>
              <b-badge class="m-1" variant="dark">Dark</b-badge>
            </div>
          </b-card>
          <b-card
            header-tag="header"
            footer-tag="footer">
            <div slot="header"><i class='fa fa-align-justify'></i> Badge <small>pills</small></div>
            <div>
              <b-badge class="m-1" pill variant="primary">Primary</b-badge>
              <b-badge class="m-1" pill variant="secondary">Secondary</b-badge>
              <b-badge class="m-1" pill variant="success">Success</b-badge>
              <b-badge class="m-1" pill variant="danger">Danger</b-badge>
              <b-badge class="m-1" pill variant="warning">Warning</b-badge>
              <b-badge class="m-1" pill variant="info">Info</b-badge>
              <b-badge class="m-1" pill variant="light">Light</b-badge>
              <b-badge class="m-1" pill variant="dark">Dark</b-badge>
            </div>
          </b-card>
          <b-card
            header-tag="header"
            footer-tag="footer">
            <div slot="header"><i class='fa fa-align-justify'></i> Badge <small>actionable</small></div>
            <div>
              <b-badge class="m-1" href="#" variant="primary">Primary</b-badge>
              <b-badge class="m-1" href="#" variant="secondary">Secondary</b-badge>
              <b-badge class="m-1" href="#" variant="success">Success</b-badge>
              <b-badge class="m-1" href="#" variant="danger">Danger</b-badge>
              <b-badge class="m-1" href="#" variant="warning">Warning</b-badge>
              <b-badge class="m-1" href="#" variant="info">Info</b-badge>
              <b-badge class="m-1" href="#" variant="light">Light</b-badge>
              <b-badge class="m-1" href="#" variant="dark">Dark</b-badge>
            </div>
          </b-card>
        </b-col>
      </b-row>
    </div>

  </div>

</template>

<script>
export default {
  name: 'badges',
  data () {
    return {
    }
  }
}
</script>
