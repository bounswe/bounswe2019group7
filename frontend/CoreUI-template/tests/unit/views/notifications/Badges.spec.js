import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Badges from '@/views/notifications/Badges'

Vue.use(BootstrapVue)

describe('Badges.vue', () => {
  it('has a name', () => {
    expect(Badges.name).toMatch('badges')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Badges)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Badges', () => {
    const wrapper = shallowMount(Badges)
    expect(wrapper.is(Badges)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Badges)
    expect(wrapper.find('header.card-header > div').text()).toMatch('Bootstrap Badge docs')
  })
  test('renders correctly', () => {
    const wrapper = mount(Badges)
    expect(wrapper.element).toMatchSnapshot()
  })
})
