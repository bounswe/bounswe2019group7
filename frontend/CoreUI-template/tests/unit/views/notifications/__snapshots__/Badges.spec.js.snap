// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges.vue renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="animated fadeIn"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
              <strong>
                 Bootstrap Badge
              </strong>
               
              <div
                class="card-header-actions"
              >
                <a
                  class="card-header-action"
                  href="https://bootstrap-vue.js.org/docs/components/badge"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <small
                    class="text-muted"
                  >
                    docs
                  </small>
                </a>
              </div>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <h2>
                Example heading 
                <span
                  class="badge badge-secondary"
                >
                  New
                </span>
              </h2>
               
              <h3>
                Example heading 
                <span
                  class="badge badge-secondary"
                >
                  New
                </span>
              </h3>
               
              <h4>
                Example heading 
                <span
                  class="badge badge-secondary"
                >
                  New
                </span>
              </h4>
               
              <h5>
                Example heading 
                <span
                  class="badge badge-secondary"
                >
                  New
                </span>
              </h5>
               
              <h6>
                Example heading 
                <span
                  class="badge badge-secondary"
                >
                  New
                </span>
              </h6>
            </div>
             
          </div>
          <footer
            class="card-footer"
          >
            <div>
              <button
                class="btn btn-primary"
                type="button"
              >
                
              Notifications 
                <span
                  class="badge badge-light"
                  style="position: static;"
                >
                  4
                </span>
              </button>
            </div>
          </footer>
          <!---->
        </div>
      </div>
       
      <div
        class="col-md-6 col-12"
      >
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
               Badge 
              <small>
                contextual variations
              </small>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <span
                class="badge m-1 badge-primary"
              >
                Primary
              </span>
               
              <span
                class="badge m-1 badge-secondary"
              >
                Secondary
              </span>
               
              <span
                class="badge m-1 badge-success"
              >
                Success
              </span>
               
              <span
                class="badge m-1 badge-danger"
              >
                Danger
              </span>
               
              <span
                class="badge m-1 badge-warning"
              >
                Warning
              </span>
               
              <span
                class="badge m-1 badge-info"
              >
                Info
              </span>
               
              <span
                class="badge m-1 badge-light"
              >
                Light
              </span>
               
              <span
                class="badge m-1 badge-dark"
              >
                Dark
              </span>
            </div>
          </div>
          <!---->
          <!---->
        </div>
         
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
               Badge 
              <small>
                pills
              </small>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <span
                class="badge m-1 badge-primary badge-pill"
              >
                Primary
              </span>
               
              <span
                class="badge m-1 badge-secondary badge-pill"
              >
                Secondary
              </span>
               
              <span
                class="badge m-1 badge-success badge-pill"
              >
                Success
              </span>
               
              <span
                class="badge m-1 badge-danger badge-pill"
              >
                Danger
              </span>
               
              <span
                class="badge m-1 badge-warning badge-pill"
              >
                Warning
              </span>
               
              <span
                class="badge m-1 badge-info badge-pill"
              >
                Info
              </span>
               
              <span
                class="badge m-1 badge-light badge-pill"
              >
                Light
              </span>
               
              <span
                class="badge m-1 badge-dark badge-pill"
              >
                Dark
              </span>
            </div>
          </div>
          <!---->
          <!---->
        </div>
         
        <div
          class="card"
        >
          <!---->
          <header
            class="card-header"
          >
            <div>
              <i
                class="fa fa-align-justify"
              />
               Badge 
              <small>
                actionable
              </small>
            </div>
          </header>
          <div
            class="card-body"
          >
            <!---->
            <!---->
             
            <div>
              <a
                class="badge m-1 badge-primary"
                href="#"
                target="_self"
              >
                Primary
              </a>
               
              <a
                class="badge m-1 badge-secondary"
                href="#"
                target="_self"
              >
                Secondary
              </a>
               
              <a
                class="badge m-1 badge-success"
                href="#"
                target="_self"
              >
                Success
              </a>
               
              <a
                class="badge m-1 badge-danger"
                href="#"
                target="_self"
              >
                Danger
              </a>
               
              <a
                class="badge m-1 badge-warning"
                href="#"
                target="_self"
              >
                Warning
              </a>
               
              <a
                class="badge m-1 badge-info"
                href="#"
                target="_self"
              >
                Info
              </a>
               
              <a
                class="badge m-1 badge-light"
                href="#"
                target="_self"
              >
                Light
              </a>
               
              <a
                class="badge m-1 badge-dark"
                href="#"
                target="_self"
              >
                Dark
              </a>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
